workspaces/liferay-ticket-workspace/client-extensions/liferay-ticket-custom-element/src/services/tickets.ts

```diff
@@ -123,3 +125,48 @@ export async function generateNewTicket() {
 		method: 'POST',
 	});
 }
+
+export async function updateTicketStatus(ticket: Ticket) {
+	let listTypeDefinitions = {} as ListTypeDefinitions;
+
+	if (!(J3Y7_PRIORITIES in listTypeDefinitions)) {
+		listTypeDefinitions = await fetchListTypeDefinitions();
+	}
+	const priorities = listTypeDefinitions[J3Y7_PRIORITIES] as any[];
+	const regions = listTypeDefinitions[J3Y7_REGIONS] as any[];
+	const resolutions = listTypeDefinitions[J3Y7_RESOLUTIONS] as any[];
+	const types = listTypeDefinitions[J3Y7_TYPES] as any[];
+	const statuses = listTypeDefinitions[J3Y7_STATUSES] as any[];
+
+	ticket.priority = priorities.find((p) => p.name === ticket.priority);
```

Comment:
We shouldn't need to do a find for all of these. If you instead have an object for each of these where the key is the name and the value is the entity, then getting the entity is just a direct lookup.

All of this prep related to the api call should only happen once. Then the data should be stored in memory and used directly rather than refetching and processing any of it.